.github/workflows/main.yml

```diff
@@ -67,6 +67,13 @@ jobs:
     - run: cargo test --features 'deny-warnings' -p cargo-test-support
     - run: cargo test -p cargo-platform
     - run: cargo test --manifest-path crates/mdman/Cargo.toml
+    - run: cargo build --manifest-path crates/credential/cargo-credential-1password/Cargo.toml
+    - run: cargo build --manifest-path crates/credential/cargo-credential-gnome-secret/Cargo.toml
+      if: matrix.os == 'ubuntu-latest'
+    - run: cargo build --manifest-path crates/credential/cargo-credential-macos-keychain/Cargo.toml
+      if: matrix.os == 'macos-latest'
+    - run: cargo build --manifest-path crates/credential/cargo-credential-wincred/Cargo.toml
+      if: matrix.os == 'windows-latest'
```

Comment:
I'm only just starting, but for convenience it'd be nice if these crates built on all platforms and then at runtime just returned errors on the wrong platform, that way things like `cargo test --all` would work well.

Reply:
I'm a little uncomfortable doing that since for example the gnome package won't work if libsecret isn't installed, and I think it could cause confusing problems if it silently ignored the absence of that library. 

What do you think about adding that kind of logic to Cargo? That is, you could somehow specify a binary only works on specific targets, and is otherwise ignored (similar to required-features). I suspect it will be a long while before Cargo can use a workspace (due to the lack of nested workspaces), so I imagine this won't be a concern for a while.

If you'd really prefer them to always compile, I'm fine with doing that, I just feel it could lead to confusing problems.

Reply:
Nah it's not a strong preference on my part, mostly just cleaning up CI config and ideally making integration elsewhere easier. (rustbuild will need to encode these platform rules, right?)

 FWIW I was imagining that we wouldn't silently ignore the lack of something like libsecret, but we would ignore winapi when building for Linux for example. As for target-specific binaries, seems like a reasonable feature to me to add!